sdk/azidentity/azure_pipelines_credential.go

```diff
@@ -86,6 +88,8 @@ func NewAzurePipelinesCredential(tenantID, clientID, serviceConnectionID, system
 	if options == nil {
 		options = &AzurePipelinesCredentialOptions{}
 	}
+	// these headers are useful to the DevOps team when debugging OIDC error responses
+	options.ClientOptions.Logging.AllowedHeaders = append(options.ClientOptions.Logging.AllowedHeaders, xMsEdgeRef, xVssE2eId)
```

Comment:
Do these options apply to the request that's sent to the OIDC endpoint?

It looks like they are being passed into `ClientAssertionCredentialOptions` (only), which means the second request will get the options, but if you have a separate pipeline and client for the OIDC request, it may not get the options (`AzurePipelinesCredential`).

Reply:
Yes (a test covers this). OIDC and Entra requests transit the same pipeline